src/Core/Utils.fs

```diff
@@ -1,7 +1,16 @@
 namespace Ionide.VSCode.FSharp
 
 open System
-open Fable.Import.vscode
+open Fable.Import.VSCode
+open Fable.Import.VSCode.Vscode
+open Fable.Core
+
+[<AutoOpen>]
+module PromiseBuilder =
+    type Foo = JS.PromiseConstructor
+    type Promise.PromiseBuilder with
+        member x.Source(p: Thenable<'t>): JS.Promise<'t> = box p |> unbox
```

Comment:
adding these lets us await Thenables.  we also have to add the self-referential Source overload or else overload resolution fails.  this 'default' Source should be contributed upstream.